File: solution/409/solution.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int charmap[128] = {0};
        int flag = 0;
        int maxlegth=0;
        for(int i=0;i<s.length();i++){
            charmap[s[i]]++;
        }
        for(int i=0;i<128;i++){
            if(charmap[i]%2==0){
                maxlegth += charmap[i];
            }else{
                flag=1;
                maxlegth += charmap[i] - 1;
            }
        }
        return maxlegth+flag;
    }
};

File: solution/409/solution.py
from collections import defaultdict as dfdict

class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        charmap = dfdict(int)
        maxlength = 0
        flag = 0
        for x in s:
            charmap[x] += 1
        for k,v in charmap.items():
            if v % 2 == 0:
                maxlength += v
            else:
                flag = 1
                maxlength += v-1
        return maxlength + flag


s = Solution()
print s.longestPalindrome('abccccdd')